ros2_batch_job/__main__.py

```diff
@@ -654,6 +654,11 @@ def run(args, build_function, blacklisted_package_names=None):
                 info('Attempting to create a well known branch name for all the default branches')
                 job.run(vcs_cmd + ['custom', '.', '--git', '--args', 'checkout', '-b', '__ci_default'])
 
+                # Check if the branch exists in at least one repo from the given repos file
+                for filename in repos_filenames:
+                    branch_found = check_branch_exists(filename, args.test_branch)
+                    if not branch_found:
+                        print('Warning: None of the repos in ', filename, ' contain the branch ', args.test_branch)
```

Comment:
How visible is this warning? Does it make the job yellow or do we have to notice it in the log?

IMO, it might be better to just abort the job early since it seems most likely not what the user intended and continuing with the build is probably not worth the time.

Reply:
Agreed; if the user specified a branch, but we couldn't find any of them, let's just abort the build early.  It's probably not what was intended.  We also have to be careful to succeed if the branch name is empty; that is the "normal" state of affairs, in which case we just want to continue on.